recipes/imagemagick/all/conanfile.py

```diff
@@ -100,336 +145,259 @@ def configure(self):
-                self.copy(
-                    pattern="*.h",
-                    dst=os.path.join(
-                        "include",
-                        "ImageMagick-%s" % tools.Version(self.version).major,
-                        module,
-                    ),
-                    src=os.path.join(self._source_subfolder, module),
-                )
+        autotools = Autotools(self)
+        if microsoft.is_msvc(self):
+            # FIXME: On Windows, Autotools sets DESTDIR as a Windows path
+            autotools.install(
+                [f"DESTDIR={microsoft.unix_path(self, self.package_folder)}"])
+        else:
+            autotools.install()
+
+        # remove undesired files
+        files.rmdir(self, os.path.join(self.package_folder, "lib",
+                                       "pkgconfig"))  # pc files
+        files.rmdir(self, os.path.join(self.package_folder, "etc"))
+        files.rmdir(self, os.path.join(self.package_folder, "share"))
+        files.rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+
+        if microsoft.is_msvc(self) and self.options.shared:
+            for m in self._modules:
+                files.rename(
+                    self,
+                    os.path.join(self.package_folder, "lib",
+                                 f"{self._libname(m)}.dll.lib"),
+                    os.path.join(self.package_folder, "lib",
+                                 f"{self._libname(m)}.lib"))
+
+        files.copy(self,
+                   pattern="LICENSE",
+                   dst=os.path.join(self.package_folder, "licenses"),
+                   src=self.source_folder)
 
     def _libname(self, library):
-        if self._is_msvc:
-            infix = "DB" if self.settings.build_type == "Debug" else "RL"
-            return "CORE_%s_%s_" % (infix, library)
-        else:
-            suffix = "HDRI" if self.options.hdri else ""
-            return "%s-%s.Q%s%s" % (
-                library,
-                tools.Version(self.version).major,
-                self.options.quantum_depth,
-                suffix,
-            )
+        suffix = "HDRI" if self.options.hdri else ""
+        return "{}-{}.Q{}{}".format(
+            library,
+            scm.Version(self.version).major,
+            self.options.quantum_depth,
+            suffix,
+        )
 
     def package_info(self):
         # FIXME model official FindImageMagick https://cmake.org/cmake/help/latest/module/FindImageMagick.html
         bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: {}".format(bin_path))
+        self.output.info(f"Appending PATH environment variable: {bin_path}")
         self.env_info.PATH.append(bin_path)
 
+        self.cpp_info.names["cmake_find_package_multi"] = "ImageMagick"
+        self.cpp_info.set_property("cmake_file_name", "ImageMagick")
+        self.cpp_info.set_property("cmake_target_name", "ImageMagick")
```

Comment:
Should this be `ImageMagick::ImageMagick`?

Reply:
I actually think that it should be removed, since there is no ImageMagick target. You link against one of MagickCore, Magick++ or MagickWand.

Reply:
That's what I was looking for, thanks! I would leave it out of the recipe, then, though unfortunately I think a default might still exist.